test/test_xoptional_assembly.cpp

```diff
@@ -347,6 +347,7 @@ namespace xt
         row_major_result<> rm;
         dyn_opt_ass_type vec;
         vec.resize(rm.m_shape, layout_type::row_major);
+        vec.fill(123);
```

Comment:
this stuff made valgrind complain about conditional jumps depending on uninitialized memory.